### PR TITLE
INFINITY-1680 Remove Kafka framework specialized upgrade podless -> pod

### PR DESCRIPTION
**Pull request by paegun**
* kafka: removed specialized upgrade, previously required for the podless- to pod-based- frameworks.

### CODE REVIEW

**Comment by paegun** [comment]
retest this please

**Comment by gabrielhartmann** [comment]
retest this please

**Comment by gabrielhartmann** [comment]
@paegun The upgrade test for Kafka will probably need to be updated too (`test_upgrade`).  It will be trying to do an upgrade which this PR removes.  Upgrades should now go from `beta-kafka` to the `kafka` in the repository.

**Comment by mbdcos** [comment]
CI error? frameworks/kafka/tests/test_shakedown_basics.py::test_suppress ERROR

**Comment by mbdcos** [comment]
You will need to add downgrade test probably cc: @susanxhuynh

**Comment by mbdcos** [comment]
Please take a look at PR # 766

**Comment by mbdcos** [comment]
" previously required for the podless- to pod-based- frameworks." 

I did not understand what this means?

**Comment by paegun** [comment]
@gabrielhartmann I've updated the Kafka sanity and soak tests to use the framework-general test for this functionality.

BUT the framework general functionality appears to have a bug wrt serialization, throwing the following attempting to deserialize, specifically on pod-specs:
```
*** snip ***
I0612 18:39:37.086760     4 exec.cpp:162] Version: 1.4.0
I0612 18:39:37.097367    10 exec.cpp:237] Executor registered on agent 870bffae-011a-4e0c-9386-da94b8ebb7a9-S4
Exception in thread "main" java.lang.IllegalStateException: com.mesosphere.sdk.config.ConfigStoreException: Failed to deserialize DefaultServiceSpecification from JSON: Could not resolve type id 'PortsSpec' into a subtype of [simple type, class com.mesosphere.sdk.specification.ResourceSpec]: known type ids = [DefaultResourceSpec, DefaultVolumeSpec, NamedVIPSpec, PortSpec, ResourceSpec]
 at [Source: {
  "name" : "kafka",
*** snip ***
  "replacement-failure-policy" : null
}; line: 52, column: 11] (through reference chain: com.mesosphere.sdk.specification.DefaultServiceSpec["pod-specs"]->java.util.ArrayList[0]->com.mesosphere.sdk.specification.DefaultPodSpec["task-specs"]->java.util.ArrayList[0]->com.mesosphere.sdk.specification.DefaultTaskSpec["resource-set"]->com.mesosphere.sdk.specification.DefaultResourceSet["resource-specifications"]->java.util.ArrayList[2])
	at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:148)
*** snip ***
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2726)
	at com.mesosphere.sdk.config.SerializationUtils.fromString(SerializationUtils.java:145)
	at com.mesosphere.sdk.specification.DefaultServiceSpec$ConfigFactory.parse(DefaultServiceSpec.java:296)
```

**Comment by paegun** [comment]
@mbdcos Yes, the CI error was cleared up by waiting and merging master. Otherwise said, the master that this was branched from wasn't clean.

@mbdcos By "... podless- to pod-based- frameworks." I mean the framework versions that did not support pods (1.8) to the versions that do support pods (1.9+). The version numbers here represent the coinciding DC/OS version that the framework version is/was released w/i. The state of things that the team was reconciling follows:

1. only worked correctly for a major version upgrade from 1.8 to 1.9 and not a minor version upgrade w/i 1.9
1. frameworks must (read: should only) support a single major version upgrade, 1.8 is off the table
1. frameworks should support minor version upgrade

The reconciliation was to remove the specialized upgrade code from the Kafka framework and assure that the upgrade/downgrade tests pass in 1.10, so testing 1.9 <-> 1.10 .

**Comment by gabrielhartmann** [comment]
@mbdcos:  We've been directed to abandon the in place upgrade work you did.

File: frameworks/kafka/tests/test_upgrade.py [hunk 1]
```diff
@@ -28,58 +19,23 @@ def teardown_module(module):
     install.uninstall(SERVICE_NAME)
 
 
+@pytest.mark.skip(reason="INFINITY-1734")
+@pytest.mark.upgrade
```
Comment (gabrielhartmann) [inline frameworks/kafka/tests/test_upgrade.py]: There should be a couple PRs that can go in today, that will allow this to be enabled.

In particular this PR and this line, re-enables this test: https://github.com/mesosphere/dcos-commons/pull/1122/files#diff-2640c88535e40fe119dc4850bd2638d2L33

**Comment by mbdcos** [comment]
@gabrielhartmann  I assume we have at least one version with auto-upgrade code, after dcos-kafka-service:0.1.19  Beta-kafka had auto-upgrade code, since  it will be the next release at least one version 0.1.20 for example will have auto-upgrade.

@benclarkwood  @gabrielhartmann What release versions of Kafka we have in stub universe at the moment?

**Comment by benclarkwood** [comment]
beta-kafka has versions 1.1.21 - 1.1.24 in universe

**Pull request merged**